src/file_pipe_log/format.rs

```diff
@@ -80,23 +80,44 @@ pub(super) fn lock_file_path<P: AsRef<Path>>(dir: P) -> PathBuf {
 }
 
 /// Version of log file format.
-#[derive(Clone, Copy, FromPrimitive, ToPrimitive)]
+#[derive(Clone, Copy, Debug, Eq, PartialEq, FromPrimitive, ToPrimitive)]
 #[repr(u64)]
-enum Version {
+pub enum Version {
     V1 = 1,
 }
 
+impl Default for Version {
+    fn default() -> Self {
+        Version::V1
+    }
+}
+
 /// In-memory representation of the log file header.
-pub(super) struct LogFileHeader {
+#[derive(Clone, Default)]
+pub struct LogFileHeader {
     version: Version,
 }
 
-impl Default for LogFileHeader {
-    fn default() -> Self {
-        Self {
-            version: Version::V1,
+impl LogFileHeader {
+    #[allow(dead_code)]
+    pub fn new(version: u64) -> Self {
+        if let Some(v) = Version::from_u64(version) {
+            Self { version: v }
+        } else {
+            Self {
+                version: Version::default(),
+            }
         }
     }
+
+    pub fn from_version(version: Version) -> Self {
+        Self { version }
+    }
+
+    #[allow(dead_code)]
+    pub fn version(&self) -> Version {
+        self.version
+    }
 }
 
 impl LogFileHeader {
```

Comment:
You are breaking the implementations into two parts again.

Reply:
Done